Test/Integration/Category/CategoryCacheTest.php

```diff
@@ -0,0 +1,341 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Test\Integration\Category;
+
+use Magento\Framework\App\Cache\Manager as CacheManager;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Framework\App\Response\Http as ResponseHttp;
+use Magento\Framework\Indexer\IndexerRegistry;
+use Magento\Store\Model\ScopeInterface;
+use Magento\TestFramework\ObjectManager;
+use Magento\TestFramework\TestCase\AbstractController;
+
+class CategoryCacheTest extends AbstractController
```

Comment:
I fear that if we start to mix every type of tests in our current directory structure, it's going to be hard to maintain. What do you think about creating folders `Test/Integration/Indexing` where we can put every existing test related to indexing , and `Test/Integration/Frontend` where we could put this test and the existing `SearchTest.php `for example ?

Reply:
This is an excellent idea! I'll reorganize this a bit.

Reply:
I went and reorganized both the integration and unit tests...

`Integration\Indexing`
![image](https://github.com/user-attachments/assets/8dd5cf2e-0d74-4ba9-8a3a-8612e4963e68)

`Integration\Frontend`
![image](https://github.com/user-attachments/assets/62fbea11-deab-45f9-9c63-fac8ea4d9aae)

`Unit`
![image](https://github.com/user-attachments/assets/e3515c15-af11-465d-95c9-b87d93659876)